R/format-report.R

```diff
@@ -307,11 +307,43 @@ format_metadata <- function(metadata_list){
     flextable::align(align = "center", part = "all") %>%
     flextable::color(color = "black", part = "body") %>%
     flextable::set_header_labels(Category = "Category", Value = "Value") %>%
-    flextable::set_caption("System Information")
+    flextable::set_caption("Execution and Machine Information")
 
   return(system_info_flextable)
 }
 
+# `cat_dependency_versions` would be a more fitting name (given cat() and
+# format()), but "format_" is used for consistency with other functions in this
+# file.
+format_dependency_versions <- function(df) {
+  out <- prepare_dependency_versions(df)
+  if (inherits(out, "flextable")) {
+    # Note: knit_print.flextable() does _not_ print to stdout.
+    out <- knitr::knit_print(out)
+  }
+  cat("\n\n", out, "\n\n")
+}
+
```

Comment:
I dont have a strong opinion on this, and would probably leave it as-is if it were up to me, but just wanted to highlight this section for @seth127 from a formatting perspective: do we want to add a minor section header for this new table?

Only reason I ask is because when R CMD Check failed, I skimmed past the the text saying it failed, and thought I didnt have the latest changes for a second:
> "Unable to calculate R dependency table due to failing `R CMD check`."

As I said im ok to leave as is, but just bringing it up while the PR is open

Reply:
@seth127 FWIW here are two examples you can reference:

### Passing R CMD Check:
![image](https://github.com/metrumresearchgroup/mpn.scorecard/assets/18685116/5889d83e-732b-411b-9ff7-044838dcc687)


### Failing R CMD Check
 - included top of R CMD Check results just to illustrate the continuity a bit more here
![image](https://github.com/metrumresearchgroup/mpn.scorecard/assets/18685116/f6cbb67b-d45d-4784-9ccf-3ebeb7263c73)